app/src/main/java/jp/developer/retia/frozenword/ui/watch/WatchViewModel.kt

```diff
@@ -0,0 +1,60 @@
+package jp.developer.retia.frozenword.ui.watch
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.viewModelScope
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedFactory
+import dagger.assisted.AssistedInject
+import java.util.Date
+import jp.developer.retia.frozenword.model.Habbit
+import jp.developer.retia.frozenword.repository.HabbitRepository
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+
+class WatchViewModel @AssistedInject constructor(
+    @Assisted private val habbitId: Int,
+    private val habbitRepository: HabbitRepository,
+    private val ioDispatcher: CoroutineDispatcher
+) : ViewModel() {
+    @AssistedFactory
+    interface ViewModelAssistedFactory {
+        fun create(habbitId: Int): WatchViewModel
+    }
+
+    private val _uiState = MutableStateFlow<WatchUiState>(WatchUiState.NotLoaded)
+    val uiState: StateFlow<WatchUiState> = _uiState.asStateFlow()
+
+    init {
+        viewModelScope.launch(ioDispatcher) {
+            val habbit = habbitRepository.getHabbit(habbitId)
+            _uiState.emit(WatchUiState.Loaded(habbit))
+        }
+    }
+
+    fun onCompleted() {
+        viewModelScope.launch(ioDispatcher){
```

Comment:
```suggestion
        viewModelScope.launch(ioDispatcher) {
```



<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>